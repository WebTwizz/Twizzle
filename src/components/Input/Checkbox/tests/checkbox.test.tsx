import React from "react";
import { render, screen } from "@testing-library/react";
import { Checkbox } from "..";

describe("Checkbox", () => {
  it("should render correctly", () => {
    const { container } = render(<Checkbox />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render with label", () => {
    const { getByText } = render(<Checkbox label="Checkbox" />);
    expect(getByText("Checkbox")).toBeInTheDocument();
  });

  it("should be disabled if given the disabled prop", () => {
    render(<Checkbox label="Checkbox" disabled />);
    expect(screen.getByRole("checkbox")).toBeDisabled();
  });

  it("should be checked if given the checked prop", () => {
    render(<Checkbox label="Checkbox" checked />);
    expect(screen.getByRole("checkbox")).toBeChecked();
  });

  it("should call onCheck if given the onCheck prop", () => {
    const onCheck = jest.fn();
    render(<Checkbox label="Checkbox" onCheck={onCheck} />);
    screen.getByRole("checkbox").click();
    expect(onCheck).toHaveBeenCalled();
  });
});
