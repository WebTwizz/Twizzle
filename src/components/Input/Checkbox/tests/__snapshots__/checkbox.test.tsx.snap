// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
<div
  id="twizzle-checkbox-container"
  style="display: flex; align-items: center; padding: 10px;"
>
  <input
    id="twizzle-checkbox"
    style="background-color: yellow; margin-right: 10px;"
    type="checkbox"
  />
  <div
    class="twizzle-typography"
    contenteditable="false"
    style="cursor: pointer; font-style: normal; font-family: sans-serif; font-weight: normal; color: black;"
  >
    <p
      style="font-size: 0.875rem;"
    >
       
       
    </p>
  </div>
</div>
`;
